sdk/storage/azure_storage_blob/src/lib.rs

```diff
@@ -3,13 +3,15 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 // Code generated by Microsoft (R) Rust Code Generator. DO NOT EDIT.
 
+pub mod clients;
 mod generated;
+pub mod pipeline;
 
-pub use crate::generated::clients::*;
+pub mod models;
 
-pub mod models {
-    pub use crate::generated::enums::*;
-    pub use crate::generated::models::*;
-}
+pub use crate::generated::clients::*;
+pub use blob_client::BlobClient as GeneratedBlobClient;
```

Comment:
Why expose your generated crate separately? If you're trying to merely extend the generated client, use extension methods as described in the guidelines: https://azure.github.io/azure-sdk/rust_implementation.html#rust-client-convenience

Reply:
wrt line 12- this line is here to support the generated code as currently imports look like:
[use crate::blob_service_client::BlobServiceClient](https://github.com/Azure/azure-sdk-for-rust/blob/d69577120fc0e1dc8b0a5471791114eb29656725/sdk/storage/azure_storage_blob/src/generated/clients/blob_client.rs#L11);
So I will need to get in touch with Joel/Catalina regarding improving the imports in generated code.

wrt line 13- That `BlobClient` is the top-level navigation client (that allows us to get the actual generated REST clients). I can change to using this in-line in the imports:
`use crate::{
    generated::clients::blob_client::BlobClient as GeneratedBlobClient,.....`

The main crux of the issue is that the top-level navigation clients struct collides with our name `BlobClient`

Reply:
My comment is really more about this line declaring it as `pub`. If you're wrapping the generated client it shouldn't be exposed as well. Whatever is exposed publicly from `src/lib.rs` is part of the public API. That's transitive throughout imports all the way up to `src/lib.rs`.